### PR TITLE
SUPPORT-3-database-handling

### PR DESCRIPTION
**Pull request by yqni13**

### CODE REVIEW

File: backend/src/controllers/meta.controller.ts [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
+import { checkValidation } from "../middleware/validation.middleware";
+import { MetaFindDTO, MetaResponseDTO, MetaUpdateDTO } from "../dtos/meta.dto";
+import { IRepoError } from "../repositories/interfaces/base.repository.interface";
 
 class MetaController {
     async getMetaData(req: Request, res: Response, next: NextFunction) {
         try {
-            const response = await metaService.getMetaData();
+            checkValidation(req);
+            const param: MetaFindDTO = { id: +(req.params.id) };
```
Comment (yqni13) [inline backend/src/controllers/meta.controller.ts:11]: rename to either dto or id

File: backend/src/middleware/validators/metaValidator.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { body, param, ValidationChain } from 'express-validator';
+import * as CustomValidator from '../../utils/customValidator.utils';
+
+export const metaFindByIdSchema: ValidationChain[] = [
+    param('id')
+        .trim()
```
Comment (yqni13) [inline backend/src/middleware/validators/metaValidator.middleware.ts:6]: use same validation as for metaUpdateSchema

File: backend/src/utils/customValidator.utils.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export function validateVersionStructure(version: string, numOfDelimiter: number): boolean {
```
Comment (yqni13) [inline backend/src/utils/customValidator.utils.ts:1]: tests missing

File: backend/src/middleware/validators/metaValidator.middleware.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+];
+
+export const metaUpdateSchema: ValidationChain[] = [
+    param('id')
+        .isInt()
+        .withMessage('backend-invalid-id'),
+    body('app')
+        .trim()
+        .notEmpty()
+        .withMessage('data-required'),
```
Comment (yqni13) [inline backend/src/middleware/validators/metaValidator.middleware.ts:18]: 'data-required' => 'arg-required' (rename for all occurences)

File: backend/src/repositories/interfaces/base.repository.interface.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+export interface IBaseRepository<T> {
+    findById(id: string | number): Promise<T |  IRepoError | null>;
+    update(id: string | number, data: Partial<T>): Promise<T | IRepoError | null>;
+}
+
+export interface IRepoError {
```
Comment (yqni13) [inline backend/src/repositories/interfaces/base.repository.interface.ts:6]: extract to seperate file

File: backend/src/repositories/interfaces/base.repository.interface.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+export interface IFindRepository<T> {
+    findAll(): Promise<T[] | IRepoError>;
+}
+
+export interface ICreateRepository<T> {
+    create(entity: T): Promise<T | IRepoError>;
+}
+
+export interface IDeleteRepository {
+    delete(id: string): Promise<boolean | IRepoError>;
```
Comment (yqni13) [inline backend/src/repositories/interfaces/base.repository.interface.ts:20]: change type for parameter to string | number

File: backend/tests/unit-tests/repositories/meta.repositories.test.ts [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    describe('Testing valid fn calls', () => {
+
+        let sql: string;
+        beforeEach(() => {
+            sql = "SELECT * FROM meta WHERE id = $1;";
+        })
+
+        test('Params: <key> = 1', async () => {
+            const mockParam_id = 1;
+            const mockResult = {
```
Comment (yqni13) [inline backend/tests/unit-tests/repositories/meta.repositories.test.ts:28]: extract data to avoid redundancy

File: backend/tests/unit-tests/repositories/meta.repositories.test.ts [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+describe('Database tests table <meta>, priority: findById', () => {
+
+    describe('Testing valid fn calls', () => {
+
+        let sql: string;
+        beforeEach(() => {
+            sql = "SELECT * FROM meta WHERE id = $1;";
+        })
+
+        test('Params: <key> = 1', async () => {
```
Comment (yqni13) [inline backend/tests/unit-tests/repositories/meta.repositories.test.ts:26]: adapt description (add something like ", get existing entry") => similar to other test case for getting null because of non-existing entry at id 0
